src/snac/server/neo4j/Neo4JUtil.php

```diff
@@ -386,18 +386,20 @@ public function redirectConstellation(&$from, &$to) {
                     if ($relation->hasValue('arcrole'))
                         $data["arcrole"] = $relation->value('arcrole');
 
-                    // Add the relation to the other Constellation
-                    // Note: matches the two nodes, and if a relation already exists of this type (ICRELATION,
-                    //       RRELATION, HIRELATION) then it will just update that relation and overwrite any
-                    //       values in Neo4J.  If the relation doesn't exist, it will instead create the
-                    //       relation with the information.
-                    $result = $this->connector->run("MATCH (a:$startType {id: {id1} }),(b:Identity {id: {id2} })
-                                                        MERGE (a)-[r:$type]->(b) SET r += {infos}",
-                    [
-                        'id1' => $startID,
-                        'id2' => "{$to->getID()}",
-                        'infos' => $data
-                    ]);
+                    // Add the relation to the other Constellation if the ids are different
+                    if ($startID != $to->getID()) {
+                        // Note: matches the two nodes, and if a relation already exists of this type (ICRELATION,
+                        //       RRELATION, HIRELATION) then it will just update that relation and overwrite any
+                        //       values in Neo4J.  If the relation doesn't exist, it will instead create the
+                        //       relation with the information.
+                        $result = $this->connector->run("MATCH (a:$startType {id: {id1} }),(b:Identity {id: {id2} })
+                                                            MERGE (a)-[r:$type]->(b) SET r += {infos}",
+                        [
+                            'id1' => $startID,
+                            'id2' => "{$to->getID()}", // need a string for neo4j
+                            'infos' => $data
+                        ]);
+                    }
```

Comment:
I know this isn't code you were writing today, but do you think we could replace 407-413 with `$this->deleteConstellation($from)`? It might dry the code up a little, if they're functionally equivalent.

Reply:
Yes! Will do.  I think we might need to test neo4J a little bit more with the IDs.  Because in delete, it's just using the numerical value.  But in redirect, I had to use the string to get it to work.